### DISCUSSION TITLE
Ability to disable item selection of a Listbox when right clicking the mouse.

### DISCUSSION THREAD

**Discussion by manutdkid77** [AvaloniaUI/Avalonia#13340] [Q&A]
I select items in a listbox by pressing the left mouse button, however sometimes I tend to press the right mouse button which triggers a selection as well.

Is there a way in Avalonia where we can disable an item from getting selected when the right mouse button is pressed ?

Here is my listbox code

```
<ListBox Items="{Binding SearchResults}" Background="Transparent" Margin="0,15,0,0" ScrollViewer.VerticalScrollBarVisibility="Visible" Selection="{Binding Selection}" BorderThickness="1" BorderBrush="Black">
	<ListBox.Styles>
		<!-- Give the ListBoxItems a fixed padding of 0 -->
		<Style Selector="ListBoxItem">
			<Setter Property="Padding" Value="0"/>
		</Style>
	</ListBox.Styles>
</ListBox>
```

**Comment by maxkatz6** [comment]
There is no property to redefine click behavior easily, and it handles [both buttons](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/ListBoxItem.cs#L62-L63).
But you can add PointerPressed event handler to the ListBox with Tunnel strategy `listBox.AddHandler(PointerPressedEvent, Handler, Tunnel)`. Check which button was pressed on a pointer, and set e.Handler = true if it's a right button.
